multiversx_sdk_core/transaction_factories/smart_contract_transactions_factory.py

```diff
@@ -0,0 +1,174 @@
+from pathlib import Path
+from typing import Any, List, Protocol, Union
+
+from multiversx_sdk_core import Transaction
+from multiversx_sdk_core.address import Address
+from multiversx_sdk_core.code_metadata import CodeMetadata
+from multiversx_sdk_core.constants import (CONTRACT_DEPLOY_ADDRESS,
+                                           VM_TYPE_WASM_VM)
+from multiversx_sdk_core.custom_tokens import TokenComputer, TokenTransfer
+from multiversx_sdk_core.errors import BadUsageError
+from multiversx_sdk_core.interfaces import IAddress
+from multiversx_sdk_core.serializer import arg_to_string, args_to_strings
+from multiversx_sdk_core.transaction_factories.transaction_builder import \
+    TransactionBuilder
+
+
+class IConfig(Protocol):
+    chain_id: str
+    min_gas_limit: int
+    gas_limit_per_byte: int
+
+
+class SmartContractTransactionsFactory:
+    def __init__(self, config: IConfig) -> None:
+        self.config = config
+
+    def create_transaction_for_deploy(self,
+                                      sender: IAddress,
+                                      bytecode: Union[Path, bytes],
+                                      gas_limit: int,
+                                      arguments: List[Any] = [],
+                                      native_transfer_amount: int = 0,
+                                      is_upgradeable: bool = True,
+                                      is_readable: bool = True,
+                                      is_payable: bool = False,
+                                      is_payable_by_sc: bool = True) -> Transaction:
+        if isinstance(bytecode, Path):
+            bytecode = bytecode.read_bytes()
+
+        metadata = CodeMetadata(is_upgradeable, is_readable, is_payable, is_payable_by_sc)
+
+        parts = [
+            arg_to_string(bytecode),
+            arg_to_string(VM_TYPE_WASM_VM),
+            str(metadata)
+        ]
+
+        parts += args_to_strings(arguments)
+
+        transaction = TransactionBuilder(
+            config=self.config,
+            sender=sender,
+            receiver=Address.from_bech32(CONTRACT_DEPLOY_ADDRESS),
+            data_parts=parts,
+            gas_limit=gas_limit,
+            add_data_movement_gas=False,
+            amount=native_transfer_amount
+        ).build()
+
+        return transaction
+
+    def create_transaction_for_execute(self,
+                                       sender: IAddress,
+                                       contract: IAddress,
+                                       function: str,
+                                       gas_limit: int,
+                                       arguments: List[Any] = [],
+                                       native_transfer_amount: int = 0,
+                                       token_transfers: List[TokenTransfer] = []) -> Transaction:
```

Comment:
Oh, nice, so you've also implemented `transfer & execute` :rocket:

Let's though extract this new code to a separate PR, to be separately reviewed.

Reply:
Extracted this code, adjusted old code, will follow-up with a PR containing this code.